lib/AST/DiagnosticEngine.cpp

```diff
@@ -577,7 +577,21 @@ static void formatDiagnosticArgument(StringRef Modifier,
       if (shouldShowAKA(type, typeName)) {
         llvm::SmallString<256> AkaText;
         llvm::raw_svector_ostream OutAka(AkaText);
-        OutAka << type->getCanonicalType();
+
+        Type desugaredType;
+        // For aka generic function types just desugar aliases on
+        // the type.
+        if (type->is<GenericFunctionType>()) {
+          desugaredType = type.transform([](Type visitTy) -> Type {
+            if (isa<TypeAliasType>(visitTy.getPointer()))
+              return visitTy->getDesugaredType();
+            return visitTy;
+          });
+        } else {
+          desugaredType = type->getCanonicalType();
```

Comment:
Does it work if you do the desugaring transform for all types, not just generic function types?

Reply:
That was the first approach, but it just broke a lot of aka diagnostics because they were on top level only(maybe there is something else too) and also I thought that those are the generic function types which need special handling, so would make sense just to tailor the logic and continue using can type since it would be (probably) already computed. 

But, I can check this with your suggestion of doing this recursively and see what we got :)

Reply:
> Does it work if you do the desugaring transform for all types, not just generic function types?

With the new approach I think we got a nice solution where this works :)